compiler/rustc_session/src/code_stats.rs

```diff
@@ -101,6 +109,11 @@ impl CodeStats {
         self.type_sizes.borrow_mut().insert(info);
     }
 
+    pub fn record_vtable_size(&self, name: String, info: VTableSizeInfo) {
+        let prev = self.vtable_sizes.lock().insert(name.clone(), info);
+        assert!(prev.is_none(), "size of vtable for `{name}` is already recorded");
```

Comment:
This panics because paths to different traits can be the same, for example:
```rust
const _: () = {
    trait A {}
};
const _: () = {
    trait A {}
};
```

What key would be better?